tests/src/geometry_checker/testqgsgeometrychecks.cpp

```diff
@@ -1240,7 +1243,11 @@ void TestQgsGeometryChecks::testOverlapCheckToleranceBug()
   QgsFeature f;
   testContext.second[layers["overlap_layer_tolerance_bug.shp"]]->getFeature( 0, f );
   double areaOld = f.geometry().area();
+  QgsPoint pointOld_1 = f.geometry().vertexAt( 1 );
+  QgsPoint pointOld_2 = f.geometry().vertexAt( 2 );
   QCOMPARE( areaOld, 10442.710061549426 );
+  QCOMPARE( pointOld_1, QgsPoint( 2537221.53079314017668366, 1152360.02460834058001637 ) );
```

Comment:
Can you swap this to QGSCOMPARENEAR and add some tolerance to the test? Testing this high precision looks fragile to me.